setup.py

```diff
@@ -45,6 +45,8 @@
         "black==24.2.0",
         "pre-commit==2.20.0",
         "ray",
+        "shap==0.44.1",
```

Comment:
@mirand863 I'm hardcoding this version of shap since there are some major changes in the API. The format and shape of `shap_values` has changed from version 0.44.1 -> [0.45.0](https://colab.research.google.com/drive/14sNFNveE1SPQvrEEAKKUSD1JPtFW6Y9S?usp=sharing). This was the reason why the previous PR was failing.

For quick comparison on the changed shapes of shap values, checkout [this](https://colab.research.google.com/drive/14sNFNveE1SPQvrEEAKKUSD1JPtFW6Y9S?usp=sharing) colab notebook.